model/test_results.go

```diff
@@ -579,6 +586,117 @@ func FindTestResults(ctx context.Context, env cedar.Environment, opts FindTestRe
 	return results, nil
 }
 
+// FindTestSamplesOptions specifies test results to collect test names from
+// and regex filters to apply to the test names.
+type FindTestSamplesOptions struct {
+	Tasks           []FindTestResultsOptions
+	TestNameRegexes []string
+}
+
+func (opts *FindTestSamplesOptions) createFindQuery() map[string]interface{} {
+	findQueries := []bson.M{}
+	for _, task := range opts.Tasks {
+		findQueries = append(findQueries, task.createFindQuery())
+	}
+
+	return bson.M{"$or": findQueries}
+}
+
+func (opts *FindTestSamplesOptions) createFindOptions() *options.FindOptions {
+	return options.Find().SetProjection(bson.M{
+		bsonutil.GetDottedKeyName(testResultsInfoKey, testResultsInfoTaskIDKey):        1,
+		bsonutil.GetDottedKeyName(testResultsInfoKey, testResultsInfoDisplayTaskIDKey): 1,
+		bsonutil.GetDottedKeyName(testResultsInfoKey, testResultsInfoExecutionKey):     1,
+		testResultsFailedTestsSampleKey:                                                1,
+	})
+}
+
+func (opts *FindTestSamplesOptions) makeTestSamples(testResults []TestResults) []TestResultsSample {
+	samples := consolidateSamples(testResults)
+	if len(opts.TestNameRegexes) > 0 {
+		samples = filterTestNames(samples, opts.TestNameRegexes)
+	}
+
+	return samples
+}
+
+func consolidateSamples(testResults []TestResults) []TestResultsSample {
+	type taskExecutionPair struct {
+		taskID    string
+		execution int
+	}
+	taskPairMap := make(map[taskExecutionPair][]string)
+	for _, result := range testResults {
+		taskID := result.Info.DisplayTaskID
```

Comment:
Are we sure that we want to always default to using the display task id? What if an execution task id was passed in? In that case, I think we would want to return the execution task id along with the filtered failed test names.

Reply:
I rejiggered `consolidateSamples` to return what was asked for. So now if you ask for an execution task you'll get just that execution task.